sway-core/src/lib.rs

```diff
@@ -200,12 +217,16 @@ fn parse_in_memory(
     Ok((lexed_program, parsed::ParseProgram { kind, root }))
 }
 
-/// Contains the lexed and parsed submodules 'deps' of a module.
-struct Submodules {
-    lexed: Vec<(Ident, lexed::LexedSubmodule)>,
-    parsed: Vec<(Ident, parsed::ParseSubmodule)>,
+pub struct Submodule {
+    name: Ident,
+    path: ModulePath,
+    lexed: lexed::LexedSubmodule,
+    parsed: parsed::ParseSubmodule,
 }
 
+/// Contains the lexed and parsed submodules 'deps' of a module.
```

Comment:
Just out of interest, what's the reason for not also including the typed submodules?